document-store/src/main/java/org/hypertrace/core/documentstore/mongo/MongoCollection.java

```diff
@@ -132,6 +134,47 @@ public boolean upsert(Key key, Document document) throws IOException {
     }
   }
 
+  /**
+   * Bulk updates existing documents if condition for the corresponding document evaluates to true.
+   */
+  @Override
+  public BulkUpdateResult bulkUpdate(List<Triple<Key, Document, Filter>> documents)
```

Comment:
Can we create a custom class encapsulating the three fields instead of the Triple? A Triple doesn't look clean on the signature of a public method.

Reply:
something like a `BulkUpdateRequest`

Reply:
introduced BulkUpdateRequest